File: Etap 4 - Implementacja C++/MiddleAges/price.cpp
#include "price.h"

// CONSTRUCTORS
Price::Price()
{
}

// MEMBER FUNCTIONS
void Price::increasePrice(double gold, double wood, double stone, double iron, double food)
{
    priceGold += gold;
    priceWood += wood;
    priceStone += stone;
    priceIron += iron;
    priceFood += food;
}


void Price::increaseBuyPrice(double gold, double wood, double stone, double iron, double food)
{
    priceBuyGold += gold;
    priceBuyWood += wood;
    priceBuyStone += stone;
    priceBuyIron += iron;
    priceBuyFood += food;
}
// SETTERS
void Price::setPriceGold(int price)
{
    priceGold = price;
}

void Price::setPriceWood(int price)
{
    priceWood = price;
}

void Price::setPriceStone(int price)
{
    priceStone = price;
}

void Price::setPriceIron(int price)
{
    priceIron = price;
}

void Price::setPriceFood(int price)
{
    priceFood = price;
}


void Price::setPriceBuyGold(int price)
{
    priceBuyGold = price;
}

void Price::setPriceBuyWood(int price)
{
    priceBuyWood = price;
}

void Price::setPriceBuyStone(int price)
{
    priceBuyStone = price;
}

void Price::setPriceBuyIron(int price)
{
    priceBuyIron = price;
}

void Price::setPriceBuyFood(int price)
{
    priceBuyFood = price;
}

// GETTERS
double Price::getPriceGold()
{
    return priceGold;
}

double Price::getPriceWood()
{
    return priceWood;
}

double Price::getPriceStone()
{
    return priceStone;
}

double Price::getPriceIron()
{
    return priceIron;
}

double Price::getPriceFood()
{
    return priceFood;
}


double Price::getPriceBuyGold()
{
    return priceBuyGold;
}

double Price::getPriceBuyWood()
{
    return priceBuyWood;
}

double Price::getPriceBuyStone()
{
    return priceBuyStone;
}

double Price::getPriceBuyIron()
{
    return priceBuyIron;
}

double Price::getPriceBuyFood()
{
    return priceBuyFood;
}

File: Etap 5 - Implementacja C#/source code/MiddleAges/Price.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MiddleAges
{
    class Price
    {
        //#################################################
        // MEMBER FUNCTIONS
        //#################################################
        public void IncreaseUpgradePrice(double gold, double wood, double stone, double iron, double food)
        {
            _priceUpgradeGold += gold;
            _priceUpgradeWood += wood;
            _priceUpgradeStone += stone;
            _priceUpgradeIron += iron;
            _priceUpgradeFood += food;
        }

        public void IncreaseBuyPrice(double gold, double wood, double stone, double iron, double food)
        {
            _priceBuyGold += gold;
            _priceBuyWood += wood;
            _priceBuyStone += stone;
            _priceBuyIron += iron;
            _priceBuyFood += food;
        }

        //#################################################
        // GETTERS & SETTERS
        //#################################################
        public double PriceBuyGold
        {
            get { return _priceBuyGold; }
            set { _priceBuyGold = value; }
        }
        public double PriceBuyWood
        {
            get { return _priceBuyWood; }
            set { _priceBuyWood = value; }
        }
        public double PriceBuyStone
        {
            get { return _priceBuyStone; }
            set { _priceBuyStone = value; }
        }
        public double PriceBuyIron
        {
            get { return _priceBuyIron; }
            set { _priceBuyIron = value; }
        }
        public double PriceBuyFood
        {
            get { return _priceBuyFood; }
            set { _priceBuyFood = value; }
        }

        public double PriceUpgradeGold
        {
            get { return _priceUpgradeGold; }
            set { _priceUpgradeGold = value; }
        }
        public double PriceUpgradeWood
        {
            get { return _priceUpgradeWood; }
            set { _priceUpgradeWood = value; }
        }
        public double PriceUpgradeStone
        {
            get { return _priceUpgradeStone; }
            set { _priceUpgradeStone = value; }
        }
        public double PriceUpgradeIron
        {
            get { return _priceUpgradeIron; }
            set { _priceUpgradeIron = value; }
        }
        public double PriceUpgradeFood
        {
            get { return _priceUpgradeFood; }
            set { _priceUpgradeFood = value; }
        }

        //#################################################
        // MEMBER VARIABLES
        //#################################################
        private double _priceBuyGold;
        private double _priceBuyWood;
        private double _priceBuyStone;
        private double _priceBuyIron;
        private double _priceBuyFood;

        private double _priceUpgradeGold;
        private double _priceUpgradeWood;
        private double _priceUpgradeStone;
        private double _priceUpgradeIron;
        private double _priceUpgradeFood;
    }
}
